contrib/datadog/tracing/interceptor.go

```diff
@@ -242,8 +247,15 @@ func (t *tracerSpan) ForeachBaggageItem(handler func(k string, v string) bool) {
 
 func (t *tracerSpan) Finish(options *interceptor.TracerFinishSpanOptions) {
 	var opts []tracer.FinishOption
-	if err := options.Error; err != nil && !workflow.IsContinueAsNewError(err) {
+
+	err := options.Error
+	isErrEligible := err != nil &&
+		!workflow.IsContinueAsNewError(err) &&
+		t.CheckError != nil && t.CheckError(err)
```

Comment:
```suggestion
		(t.CheckError == nil || t.CheckError(err))
```

Otherwise no error will ever be tracked unless they set the option. Alternatively you could just default check error to be a default impl that checks for continue-as-new error and tell users that so if they override, they can choose whether continue-as-new is counted.